src/dsl/word/Pool.hpp

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright (C) 2023      Alex Biddulph <bidskii@gmail.com>
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated
+ * documentation files (the "Software"), to deal in the Software without restriction, including without limitation the
+ * rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to
+ * permit persons to whom the Software is furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all copies or substantial portions of the
+ * Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
+ * WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR
+ * OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+#ifndef NUCLEAR_DSL_WORD_POOL_HPP
+#define NUCLEAR_DSL_WORD_POOL_HPP
+
+#include <map>
+#include <mutex>
+#include <typeindex>
+
+#include "../../threading/ReactionTask.hpp"
+#include "../../util/ThreadPoolDescriptor.hpp"
+
+namespace NUClear {
+namespace dsl {
+    namespace word {
+
+        /**
+         * @brief
+         *  This is used to specify that this reaction should run in the designated thread pool
+         *
+         * @details
+         *  @code on<Trigger<T, ...>, Pool<PoolType>>() @endcode
+         *  This DSL will cause the creation of a new thread pool with a specific number of threads allocated to it.
+         *
+         *  All tasks for this reaction will be queued to run on threads from this thread pool.
+         *
+         *  Tasks in the queue are ordered based on their priority level, then their task id.
+         *
+         *  When this DSL is not specified the default thread pool will be used. For tasks that need to run on the main
+         *  thread use MainThread.
+         *
+         *  For best use, this word should be fused with at least one other binding DSL word.
```

Comment:
I think we can just drop this word sentence from everywhere. Everything needs a binding dsl word